pandas/core/indexing.py

```diff
@@ -1815,14 +1821,13 @@ def _setitem_single_block(self, indexer, value):
                 return
 
             indexer = maybe_convert_ix(*indexer)
-
-        if isinstance(value, (ABCSeries, dict)):
+        if isinstance(value, ABCSeries) and name != "iloc" or isinstance(value, dict):
```

Comment:
for follow-up, can you put parens where appropriate so i dont have to think about whether the "and" gets evaluated before the "or"